client/gutenberg/editor/controller.js

```diff
@@ -119,23 +118,32 @@ export const post = async ( context, next ) => {
 	const postType = determinePostType( context );
 	const isDemoContent = ! postId && has( context.query, 'gutenberg-demo' );
 
-	let state = context.store.getState();
-	let siteId = getSelectedSiteId( state );
-	if ( ! siteId ) {
-		siteId = await waitForSelectedSiteId( context );
-		state = context.store.getState();
-	}
-	const siteSlug = getSelectedSiteSlug( state );
-	const userId = getCurrentUserId( state );
+	const makeEditor = siteId => {
+		const state = context.store.getState();
+		const siteSlug = getSelectedSiteSlug( state );
+		const userId = getCurrentUserId( state );
+
+		return import( './init' ).then( ( { initGutenberg } ) => {
```

Comment:
if the load of `init` is inflight and the loader unmounts, I think we'd want a way to prevent this code from running?

Reply:
D'oh, missed this :slightly_frowning_face: